sklearn/utils/validation.py

```diff
@@ -80,7 +80,15 @@ def as_float_array(X, copy=True, force_all_finite=True):
     elif X.dtype in [np.float32, np.float64]:  # is numpy array
         return X.copy('F' if X.flags['F_CONTIGUOUS'] else 'C') if copy else X
     else:
-        return X.astype(np.float32 if X.dtype == np.int32 else np.float64)
+        return X.astype(np.float32 if X.dtype in (
+                                                  np.int8,
```

Comment:
Not sure what you mean, do you have a stand-alone snippet where this problem happens?